src/main/java/org/mifos/processor/bulk/camel/routes/ProcessorStartRoute.java

```diff
@@ -233,19 +267,158 @@ private void setup() {
+                    "The file uploaded is broken as it has a different extension. " +
+                            "Please upload a csv file and try again.");
+                    }
+                 else{
+                     setResponse(exchange, 200);}
+
+    });}
+
+    private boolean verifyData(File file) throws IOException {
+        InputStream ips = new FileInputStream(file);
+        InputStreamReader ipsr = new InputStreamReader(ips);
+        BufferedReader br = new BufferedReader(ipsr);
+        String line;
+        br.readLine();
+        while ((line = br.readLine()) != null) {
+            String[] row = line.split(",");
+            logger.info("row is" + Arrays.toString(row));
+            if (row.length != columnNames.size()) {
+                logger.info("Row invalid " + row.length + " " + columnNames.size());
+                return false;
+            }
+            if(!verifyRow(row))
+                return false;
+        }
+        return true;
+    }
+    private boolean verifyRow(String[] row) {
+            for (int i = 1; i < row.length; i++) {
+                row[i] = row[i].trim();
+                if (row[i].equalsIgnoreCase("MSISDN")) {
+                    logger.info("reached here");
+                    if (!row[i+ 1].matches("^[0-9]*$")) {
+                        logger.info("MSISDN invalid");
+                        return false;
+                    }
+                }
+                else if (row[i].contains("amount")){
+                    int j = row[i].indexOf("amount");
+                    if(!row[j].matches("^[0-9]*$")){
+                        logger.info("Amount invalid");
+                        return false;
+                    }
+
+                }
+            }
+            return true;
+        }
+
+    private File setupfile(String csvData, String fileName) throws IOException {
+        File file = new File(fileName);
+        file.setWritable(true);
+        file.setReadable(true);
+        FileWriter fileWriter = new FileWriter(file);
+        fileWriter.write(csvData);
+        fileWriter.close();
+        return file;
     }
 
+
+    private boolean verifyCsv(File csvData) throws IOException {
+        // TODO Auto-generated method stub
+        BufferedReader br = new BufferedReader(new FileReader(csvData));
+        String header = br.readLine();
+        String[] columns = new String[0];
+        if (header != null) {
+            columns = header.split(",");
+            logger.info("Columns in the csv file are " + Arrays.toString(columns));
+        }
+        int i = 0;
+        while(i < columns.length) {
+            if(columnNames.contains(columns[i])) {
+                logger.info("Column name " + columns[i] + " is at index " + columnNames.indexOf(columns[i]));
+                i++;
+
+            } else {
+                logger.info("Column name " + columns[i] + " is at index " + columnNames.indexOf(columns[i]));
```

Comment:
when contains operation is false, how are you expecting this `columnNames.indexOf(columns[i]` to work?

Reply:
this was a dev logger that i can remove now